proxy/ha_proxy.go

```diff
@@ -133,7 +133,7 @@ func (m HaProxy) Reload() error {
 		if err != nil {
 			return fmt.Errorf("Could not read the %s file\n%s", pidPath, err.Error())
 		}
-		cmdArgs := []string{"-sf", string(pid)}
+		cmdArgs := []string{"-st", string(pid)}
```

Comment:
-sf seems to be the better option and safer for most cases, but never the less it seems that some requests never finish. On my machine with -sf the process gets stuck.

One idea I have is that this might be due to swarm health checking. Or any other ideas why SIGUSR1 signal never gets fired?

What do you think of exposing this option to the user in order to configure the proxy reloading mode?